src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py

```diff
@@ -42,6 +42,24 @@ def show_subscription(subscription=None, expanded_view=None):
         return profile.get_expanded_subscription_info(subscription)
 
 
+def get_access_token(subscription=None, resource=None):
+    '''
+    get AAD token to access to a specified resource
+    :param resource: Azure resource endpoints. Default to Azure Resource Manager
+    Use 'az cloud show' command for other Azure resources
+    '''
+    profile = Profile()
+    creds, subscription, tenant = profile.get_raw_token(subscription_id=subscription,
+                                                        resource=resource)
+    return {
+        'tokenType': creds[0],
+        'accessToken': creds[1],
+        'expiresOn': creds[2]['expiresOn'],
+        'subscription': subscription,
```

Comment:
Actually after taking a look at the existing code, looks like this would return whatever you pass in.
If I pass in ID, I'll get id back. If I pass in name, I'll get name back.

I'm getting this from _params.py
```
register_cli_argument('account', 'subscription', options_list=('--subscription', '-s'), help='Name or ID of subscription.', completer=get_subscription_id_list)
```